vms/platformvm/txs/executor/standard_tx_executor.go

```diff
@@ -215,11 +221,11 @@ func (e *standardTxExecutor) CreateChainTx(tx *txs.CreateChainTx) error {
 	if err := e.backend.FlowChecker.VerifySpend(
 		tx,
 		e.state,
-		tx.Ins,
-		tx.Outs,
+		ins,
+		outs,
 		baseTxCreds,
 		map[ids.ID]uint64{
-			e.backend.Ctx.AVAXAssetID: fee,
+			e.backend.Ctx.AVAXAssetID: fee + producedAVAX,
```

Comment:
re: https://github.com/ava-labs/avalanchego/pull/3715#discussion_r2110053889

We added the `producedAVAX` to the `fee` in these calls because we wanted to have the same behavior across all the methods... But now we seem to only check for overflow when `producedAVAX` is non-zero. It feels like we are in a pretty weird place here where we included the `producedAVAX` in the calculation, but we only include it correctly when we actually need to include it.